api/src/test/java/jakarta/data/CompositeRestrictionRecordTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data;
+
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+
+class CompositeRestrictionRecordTest {
+
+
+    @Test
+    void shouldCreateCompositeRestrictionWithDefaultNegation() {
+        Restriction<String> restriction1 = new BasicRestrictionRecord<>("title", Operator.EQUAL, "Java Guide");
+        Restriction<String> restriction2 = new BasicRestrictionRecord<>("author", Operator.EQUAL, "John Doe");
+
+        CompositeRestrictionRecord<String> composite = new CompositeRestrictionRecord<>(
+                CompositeRestriction.Type.ALL,
+                List.of(restriction1, restriction2)
+        );
+
+        SoftAssertions.assertSoftly(soft -> {
+            soft.assertThat(composite.type()).isEqualTo(CompositeRestriction.Type.ALL);
+            soft.assertThat(composite.restrictions()).containsExactly(restriction1, restriction2);
+            soft.assertThat(composite.isNegated()).isFalse();
+        });
+    }
+
+    @Test
+    void shouldCreateCompositeRestrictionWithExplicitNegation() {
+        Restriction<String> restriction1 = new BasicRestrictionRecord<>("title", Operator.EQUAL, "Java Guide");
+        Restriction<String> restriction2 = new BasicRestrictionRecord<>("author", Operator.EQUAL, "John Doe");
+
+        CompositeRestrictionRecord<String> composite = new CompositeRestrictionRecord<>(
+                CompositeRestriction.Type.ANY,
+                List.of(restriction1, restriction2),
+                true
+        );
+
+        SoftAssertions.assertSoftly(soft -> {
+            soft.assertThat(composite.type()).isEqualTo(CompositeRestriction.Type.ANY);
+            soft.assertThat(composite.restrictions()).containsExactly(restriction1, restriction2);
+            soft.assertThat(composite.isNegated()).isTrue();
+        });
+    }
+
+    @Test
+    void shouldHandleEmptyRestrictions() {
```

Comment:
We might decide to make an empty CompositeRestriction into an error, but you are correct that the code is currently allowing that.

Reply:
Yes, when we got an exception we can use the same test to verify it.